subprojects/core/src/main/java/org/gradle/api/internal/project/DefaultProject.java

```diff
@@ -597,6 +597,25 @@ public int compareTo(Project otherProject) {
         return ProjectOrderingUtil.compare(this, otherProject);
     }
 
+    @Override
+    public int hashCode() {
+        return getIdentityPath().hashCode();
+    }
+
+    @Override
+    public boolean equals(Object otherProject) {
```

Comment:
💅 I'm very fond of using `that` for this particular parameter name

Reply:
Have looked at the rest of the codebase - `that` is too uncommon for gradle/gradle. Renamed to a more well-established `other`